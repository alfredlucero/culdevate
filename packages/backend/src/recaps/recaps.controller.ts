import { Response } from "express";
import { RequestWithUser } from "../interfaces/requestWithUser";
import RecapsDao from "./recaps.dao";
import { Recap } from "./recaps.model";

const RecapsController = {
  async createRecap(req: RequestWithUser, res: Response) {
    const currentUser = req.user;
    const recap: Recap = {
      ...req.body,
      userId: currentUser.id,
    };

    try {
      const createdRecap = await RecapsDao.createRecap(recap);

      res.status(201).json(createdRecap);
    } catch (err) {
      res.status(500).json({ message: "Failed to create recap." });
    }
  },

  async updateRecap(req: RequestWithUser, res: Response) {
    const currentUser = req.user;
    const recapId = req.params.recapId;
    const updatedRecap: Recap = {
      ...req.body,
      userId: currentUser.id,
    };

    try {
      const foundRecap = await RecapsDao.findRecapById(recapId);
      const isMatchingUserId = foundRecap && foundRecap.userId.toString() === currentUser.id;

      if (!foundRecap || !isMatchingUserId) {
        return res.status(404).json({ message: "Failed to find matching recap to update" });
      }

      const changedRecap = await RecapsDao.updateRecapById({ recapId, updatedRecap });

      if (changedRecap) {
        res.status(200).json(changedRecap);
      } else {
        // If by some chance it got deleted while we were trying to update it
        // we'll still get a 404 back
        res.status(404).json({ message: "Failed to find matching recap to update" });
      }
    } catch (err) {
      res.status(500).json({ message: "Failed to update recap" });
    }
  },

  async deleteRecap(req: RequestWithUser, res: Response) {
    const recapId = req.params.recapId;
    const currentUser = req.user;

    try {
      const foundRecap = await RecapsDao.findRecapById(recapId);
      const isMatchingUserId = foundRecap && foundRecap.userId.toString() === currentUser.id;

      if (!foundRecap || !isMatchingUserId) {
        return res.status(404).json({ message: "Failed to find matching recap to delete" });
      }

      const deletedRecap = await RecapsDao.removeRecapById(recapId);

      if (deletedRecap) {
        res.status(200).json(deletedRecap);
      } else {
        // If by some chance it got deleted while we were trying to delete it
        // we'll still get a 404 back
        res.status(404).json({ message: "Failed to find matching recap to delete" });
      }
    } catch (err) {
      res.status(500).json({ message: "Failed to delete recap" });
    }
  },
};

export default RecapsController;
