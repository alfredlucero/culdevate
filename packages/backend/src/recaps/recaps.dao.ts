import {
  RecapBaseModel,
  RecapWorkExperienceModel,
  RecapEducationModel,
  RecapAccomplishmentsModel,
  RecapPublicationsModel,
  RecapSkillsModel,
  RecapSideProjectsModel,
  RecapOrganizationsModel,
  RecapReferencesModel,
  RecapOtherModel,
  Recap,
} from "./recaps.model";

const RecapsDao = {
  findAllRecaps(userId: string) {
    return RecapBaseModel.find({ userId });
  },

  findRecapById(recapId: string) {
    return RecapBaseModel.findById(recapId);
  },

  findRecapByIdAndUserId({ recapId, userId }: { recapId: string; userId: string }) {
    return RecapBaseModel.findOne({ userId, _id: recapId });
  },

  createRecap(recap: Recap) {
    switch (recap.kind) {
      case "Work Experience":
        const workExperienceRecap = new RecapWorkExperienceModel(recap);
        return workExperienceRecap.save();
      case "Education":
        const educationRecap = new RecapEducationModel(recap);
        return educationRecap.save();
      case "Accomplishments":
        const accomplishmentsRecap = new RecapAccomplishmentsModel(recap);
        return accomplishmentsRecap.save();
      case "Publications":
        const publicationsRecap = new RecapPublicationsModel(recap);
        return publicationsRecap.save();
      case "Skills":
        const skillsRecap = new RecapSkillsModel(recap);
        return skillsRecap.save();
      case "Side Projects":
        const sideProjectsRecap = new RecapSideProjectsModel(recap);
        return sideProjectsRecap.save();
      case "Organizations":
        const organizationsRecap = new RecapOrganizationsModel(recap);
        return organizationsRecap.save();
      case "References":
        const referencesRecap = new RecapReferencesModel(recap);
        return referencesRecap.save();
      case "Other":
        const otherRecap = new RecapOtherModel(recap);
        return otherRecap.save();
      default:
        throw new Error("Failed to find matching recap kind to create!");
    }
  },

  updateRecapById({ recapId, updatedRecap }: { recapId: string; updatedRecap: Recap }) {
    switch (updatedRecap.kind) {
      case "Work Experience":
        return RecapWorkExperienceModel.findOneAndUpdate(
          { _id: recapId },
          { $set: updatedRecap, ...(!updatedRecap.endDate ? { $unset: { endDate: 1 } } : {}) },
          { new: true },
        );
      case "Education":
        return RecapEducationModel.findOneAndUpdate(
          { _id: recapId },
          { $set: updatedRecap, ...(!updatedRecap.endDate ? { $unset: { endDate: 1 } } : {}) },
          { new: true },
        );
      case "Accomplishments":
        return RecapAccomplishmentsModel.findOneAndUpdate({ _id: recapId }, updatedRecap, { new: true });
      case "Publications":
        return RecapPublicationsModel.findOneAndUpdate({ _id: recapId }, updatedRecap, { new: true });
      case "Skills":
        return RecapSkillsModel.findOneAndUpdate({ _id: recapId }, updatedRecap, { new: true });
      case "Side Projects":
        return RecapSideProjectsModel.findOneAndUpdate(
          { _id: recapId },
          { $set: updatedRecap, ...(!updatedRecap.endDate ? { $unset: { endDate: 1 } } : {}) },
          { new: true },
        );
      case "Organizations":
        return RecapOrganizationsModel.findOneAndUpdate(
          { _id: recapId },
          { $set: updatedRecap, ...(!updatedRecap.endDate ? { $unset: { endDate: 1 } } : {}) },
          { new: true },
        );
      case "References":
        return RecapReferencesModel.findOneAndUpdate({ _id: recapId }, updatedRecap, { new: true });
      case "Other":
        return RecapOtherModel.findOneAndUpdate({ _id: recapId }, updatedRecap, { new: true });
      default:
        throw new Error("Failed to find matching recap kind to update!");
    }
  },

  removeRecapById(recapId: string) {
    return RecapBaseModel.findByIdAndDelete({ _id: recapId });
  },
};

export default RecapsDao;
