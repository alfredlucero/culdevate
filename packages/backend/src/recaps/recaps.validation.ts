import Joi from "@hapi/joi";

// TODO: replace with yup for validation

export const MAX_BULLETPOINT_LENGTH = 1000;
export const MAX_BULLETPOINTS = 20;
export const MAX_GENERAL_LENGTH = 254;
export const MAX_URL_LENGTH = 2048;

const RecapWorkExperienceSchema = Joi.object({
  title: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  location: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  company: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  employmentType: Joi.string()
    .valid(
      "Part-Time",
      "Full-Time",
      "Self-Employed",
      "Freelance",
      "Contract",
      "Internship",
      "Apprenticeship",
      "Volunteer",
    )
    .required(),
});

const RecapEducationSchema = Joi.object({
  school: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  location: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  degree: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  fieldOfStudy: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  grade: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
});

const RecapAccomplishmentsSchema = Joi.object({
  title: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  type: Joi.string()
    .valid("Personal", "Service", "Featured", "School", "Career", "Other")
    .required(),
});

const RecapPublicationsSchema = Joi.object({
  title: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  type: Joi.string()
    .valid("Book", "Journal", "Newspaper", "Magazine", "Blog", "Other")
    .required(),
  coauthors: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  publisher: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  url: Joi.string()
    .allow("")
    .max(MAX_URL_LENGTH),
});

const RecapSkillsSchema = Joi.object({
  title: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  proficiency: Joi.string()
    .valid("Novice", "Intermediate", "Advanced", "Expert")
    .required(),
});

const RecapSideProjectsSchema = Joi.object({
  title: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  creators: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
});

const RecapOrganizationsSchema = Joi.object({
  organizationName: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  location: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  positions: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
});

const RecapReferencesSchema = Joi.object({
  company: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  title: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  phoneNumber: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
  email: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
});

const RecapOtherSchema = Joi.object({
  title: Joi.string()
    .max(MAX_GENERAL_LENGTH)
    .required(),
});

export const RecapSchema = Joi.object({
  startDate: Joi.date().optional(),
  endDate: Joi.date().optional(),
  bulletPoints: Joi.array()
    .items(Joi.string().max(MAX_BULLETPOINT_LENGTH))
    .max(MAX_BULLETPOINTS)
    .required(),
  kind: Joi.string()
    .valid(
      "Work Experience",
      "Education",
      "Accomplishments",
      "Publications",
      "Skills",
      "Side Projects",
      "Organizations",
      "References",
      "Other",
    )
    .required(),
})
  .when(Joi.object({ kind: "Work Experience" }).unknown(), {
    then: RecapWorkExperienceSchema,
  })
  .when(Joi.object({ kind: "Education" }).unknown(), {
    then: RecapEducationSchema,
  })
  .when(Joi.object({ kind: "Accomplishments" }).unknown(), {
    then: RecapAccomplishmentsSchema,
  })
  .when(Joi.object({ kind: "Publications" }).unknown(), {
    then: RecapPublicationsSchema,
  })
  .when(Joi.object({ kind: "Skills" }).unknown(), {
    then: RecapSkillsSchema,
  })
  .when(Joi.object({ kind: "Side Projects" }).unknown(), {
    then: RecapSideProjectsSchema,
  })
  .when(Joi.object({ kind: "Organizations" }).unknown(), {
    then: RecapOrganizationsSchema,
  })
  .when(Joi.object({ kind: "References" }).unknown(), {
    then: RecapReferencesSchema,
  })
  .when(Joi.object({ kind: "Other" }).unknown(), {
    then: RecapOtherSchema,
  });
