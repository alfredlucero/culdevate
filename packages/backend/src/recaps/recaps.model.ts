import mongoose from "mongoose";
import { MAX_BULLETPOINT_LENGTH, MAX_BULLETPOINTS, MAX_GENERAL_LENGTH } from "./recaps.validation";

export interface RecapBase {
  kind: RecapKind;
  startDate?: Date;
  endDate?: Date;
  bulletPoints: string[];
}

export type RecapKind =
  | "WorkExperience"
  | "Education"
  | "Accomplishments"
  | "Skills"
  | "SideProjects"
  | "Organizations"
  | "References"
  | "Other";

const recapBaseOptions = {
  discriminatorKey: "kind",
  collection: "recaps",
};

const recapBaseSchema = new mongoose.Schema(
  {
    startDate: { type: Date },
    endDate: { type: Date },
    bulletPoints: {
      type: [
        {
          type: String,
          maxlength: MAX_BULLETPOINT_LENGTH,
        },
      ],
      required: true,
      maxlength: MAX_BULLETPOINTS,
    },
  },
  recapBaseOptions,
);

export interface RecapBaseModel extends RecapBase, mongoose.Document {}

// Base Recap model from which other Recap models will inherit from based on the
// the value of the "kind" discriminator
export const RecapBaseModel = mongoose.model<RecapBaseModel>("Recap", recapBaseSchema);

// Work Experience

export interface RecapWorkExperience {
  title: string;
  location: string;
  company: string;
  employmentType: EmploymentType;
}

export type EmploymentType = "partTime" | "fullTime";

const recapWorkExperienceSchema = new mongoose.Schema({
  title: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  location: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  company: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  employmentType: { type: String, required: true, enum: ["partTime", "fullTime"] },
});

export interface RecapWorkExperienceModel extends RecapBaseModel, RecapWorkExperience {}

const RecapWorkExperience = RecapBaseModel.discriminator("WorkExperience", recapWorkExperienceSchema);

export const RecapWorkExperienceModel = mongoose.model<RecapWorkExperienceModel>("WorkExperience");

// Education

export interface RecapEducation {
  school: string;
  location: string;
  degree: string;
  fieldOfStudy: string;
  grade: string;
}

const recapEducationSchema = new mongoose.Schema({
  school: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  location: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  degree: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  fieldOfStudy: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  grade: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
});

export interface RecapEducationModel extends RecapBaseModel, RecapEducation {}

const RecapEducation = RecapBaseModel.discriminator("Education", recapEducationSchema);

export const RecapEducationModel = mongoose.model<RecapEducationModel>("Education");

// Accomplishments

export interface RecapAccomplishments {
  title: string;
  type: string;
}

const recapAccomplishmentsSchema = new mongoose.Schema({
  title: {
    type: String,
    required: true,
    maxlength: MAX_GENERAL_LENGTH,
  },
  type: {
    type: String,
    required: true,
    maxlength: MAX_GENERAL_LENGTH,
  },
});

export interface RecapAccomplishmentsModel extends RecapBaseModel, RecapAccomplishments {}

const RecapAccomplishments = RecapBaseModel.discriminator("Accomplishments", recapAccomplishmentsSchema);

export const RecapAccomplishmentsModel = mongoose.model<RecapAccomplishmentsModel>("Accomplishments");

// Skills

export interface RecapSkills {
  title: string;
}

const recapSkillsSchema = new mongoose.Schema({
  title: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
});

export interface RecapSkillsModel extends RecapBaseModel, RecapSkills {}

const RecapSkills = RecapBaseModel.discriminator("Skills", recapSkillsSchema);

export const RecapSkillsModel = mongoose.model<RecapSkillsModel>("Skills");

// Side Projects

export interface RecapSideProjects {
  title: string;
}

const recapSideProjectsSchema = new mongoose.Schema({
  title: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
});

export interface RecapSideProjectsModel extends RecapBaseModel, RecapSideProjects {}

const RecapSideProjects = RecapBaseModel.discriminator("SideProjects", recapSideProjectsSchema);

export const RecapSideProjectsModel = mongoose.model<RecapSideProjectsModel>("SideProjects");

// Organizations

export interface RecapOrganizations {
  organizationName: string;
  location: string;
  positions: string;
}

const recapOrganizationsSchema = new mongoose.Schema({
  organizationName: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  location: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  positions: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
});

export interface RecapOrganizationsModel extends RecapBaseModel, RecapOrganizations {}

const RecapOrganizations = RecapBaseModel.discriminator("Organizations", recapOrganizationsSchema);

export const RecapOrganizationsModel = mongoose.model<RecapOrganizationsModel>("Organizations");

// References

export interface RecapReferences {
  company: string;
  title: string;
  phoneNumber: string;
  email: string;
}

export const recapReferencesSchema = new mongoose.Schema({
  company: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  title: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  phoneNumber: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
  email: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
});

export interface RecapReferencesModel extends RecapBaseModel, RecapReferences {}

const RecapReferences = RecapBaseModel.discriminator("References", recapReferencesSchema);

export const RecapReferencesModel = mongoose.model<RecapReferencesModel>("References");

// Other

export interface RecapOther {
  title: string;
}

const recapOtherSchema = new mongoose.Schema({
  title: { type: String, required: true, maxlength: MAX_GENERAL_LENGTH },
});

export interface RecapOtherModel extends RecapBaseModel, RecapOther {}

const RecapOther = RecapBaseModel.discriminator("Other", recapOtherSchema);

export const RecapOtherModel = mongoose.model<RecapOtherModel>("Other");
