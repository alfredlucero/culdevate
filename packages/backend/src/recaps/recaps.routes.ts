import { Router } from "express";
import RecapsController from "./recaps.controller";
import authMiddleware from "../middleware/auth.middleware";
import validationMiddleware from "../middleware/validation.middleware";
import { RecapSchema } from "./recaps.validation";

const RecapsRoutes = {
  path: "/recaps",
  router: Router(),

  initializeRoutes() {
    // POST /recaps
    this.router.post("/", authMiddleware, validationMiddleware(RecapSchema, "body"), RecapsController.createRecap);

    // PATCH /recaps/:recapId
    this.router.patch(
      "/:recapId",
      authMiddleware,
      validationMiddleware(RecapSchema, "body"),
      RecapsController.updateRecap,
    );

    return this.router;
  },
};

export default RecapsRoutes;
