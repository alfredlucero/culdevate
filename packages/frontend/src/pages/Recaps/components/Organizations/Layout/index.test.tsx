import React from "react";
import { render, fireEvent, waitForElementToBeRemoved } from "@testing-library/react";
import OrganizationsLayout, { OrganizationsLayoutProps } from "./index";
import { RecapOrganizations } from "../../../recaps.interface";
import * as RecapsService from "../../../recaps.service";

const sampleRecapOrganizations: RecapOrganizations = {
  organizationName: "Zeta Mu Beta",
  positions: "Pledge Educator, Member",
  location: "Long Beach, CA",
  startDate: new Date("2014/12/13").toISOString(),
  endDate: new Date("2016/03/31").toISOString(),
  kind: "Organizations",
  userId: "userId",
  _id: "organizationsId",
  bulletPoints: [
    "Crossed into the organization in Fall 2014",
    "Served as an active member for 2 years and as a pledge educator for the XI class",
  ],
};
const defaultProps: OrganizationsLayoutProps = {
  recaps: [sampleRecapOrganizations],
  onGoBackToLanding: () => {},
  onCreateRecapSuccess: () => {},
  onUpdateRecapSuccess: () => {},
  onDeleteRecapSuccess: () => {},
};

describe("<OrganizationsLayout />", () => {
  test("should render without error", () => {
    const { container } = render(
      <OrganizationsLayout
        {...defaultProps}
        testId="organizationsLayoutTestId"
        className="extra-organizations-layout-class"
      />,
    );

    expect(container).toMatchSnapshot();
  });

  test("should show the empty state when there are no recaps", () => {
    const { getByTestId, queryByTestId } = render(<OrganizationsLayout {...defaultProps} recaps={[]} />);

    expect(queryByTestId("organizationsRecap")).toBeNull();

    expect(getByTestId("organizationsEmptyCard")).toBeVisible();
  });

  test("should show the recaps when there is data", () => {
    const { getByTestId, queryByTestId } = render(<OrganizationsLayout {...defaultProps} />);

    expect(queryByTestId("organizationsEmptyCard")).toBeNull();

    expect(getByTestId("organizationsRecap")).toBeVisible();
  });

  test("should show the success alert after deleting a recap successfully", async () => {
    const onDeleteRecapSuccessMock = jest.fn();
    const { getByText, queryByText, findByText } = render(
      <OrganizationsLayout {...defaultProps} onDeleteRecapSuccess={onDeleteRecapSuccessMock} />,
    );

    fireEvent.click(getByText("Delete"));

    expect(getByText("Are you sure you want to delete this?")).toBeVisible();

    jest.spyOn(RecapsService, "deleteRecap").mockImplementationOnce(() => Promise.resolve(sampleRecapOrganizations));

    fireEvent.click(getByText("Confirm Delete"));

    await waitForElementToBeRemoved(() => queryByText("Are you sure you want to delete this?"));

    expect(onDeleteRecapSuccessMock).toHaveBeenCalled();
    const successAlert = await findByText("You have successfully deleted the Organizations Recap!");
    expect(successAlert).toBeVisible();
  });

  test("should show the error alert after failing to delete a recap", async () => {
    const onDeleteRecapSuccessMock = jest.fn();
    const { getByText, queryByText, findByText } = render(
      <OrganizationsLayout {...defaultProps} onDeleteRecapSuccess={onDeleteRecapSuccessMock} />,
    );

    fireEvent.click(getByText("Delete"));

    expect(getByText("Are you sure you want to delete this?")).toBeVisible();

    jest.spyOn(RecapsService, "deleteRecap").mockImplementationOnce(() => Promise.reject({}));

    fireEvent.click(getByText("Confirm Delete"));

    await waitForElementToBeRemoved(() => queryByText("Are you sure you want to delete this?"));

    expect(onDeleteRecapSuccessMock).not.toHaveBeenCalled();
    const errorAlert = await findByText("Something went wrong with deleting the Organizations Recap!", {
      exact: false,
    });
    expect(errorAlert).toBeVisible();
  });
});
