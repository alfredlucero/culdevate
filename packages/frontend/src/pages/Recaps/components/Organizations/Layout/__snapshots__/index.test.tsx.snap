// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrganizationsLayout /> should render without error 1`] = `
<div>
  <div
    class="extra-organizations-layout-class"
    data-testid="organizationsLayoutTestId"
  >
    <header
      class="mb-8"
    >
      <button
        class="font-bold py-2 px-4 rounded bg-transparent text-teal-700 border border-teal-500 hover:bg-teal-500 hover:text-white hover:border-transparent mb-4"
        type="button"
      >
        <div
          class="flex items-center"
        >
          <div
            class="font-sans text-2xl mr-2"
            id=""
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-chevron-left fa-w-10 "
              data-icon="chevron-left"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
                fill="currentColor"
              />
            </svg>
          </div>
           Back to Recaps
        </div>
      </button>
      <div>
        <div
          class="flex justify-between w-full"
        >
          <h2
            class="font-mono font-bold text-gray-800 text-3xl "
          >
            Organizations
          </h2>
          <button
            class="font-bold py-2 px-4 rounded font-bold py-2 px-4 rounded bg-teal-500 text-white hover:bg-teal-700 "
            type="button"
          >
            <div
              class="flex items-center"
            >
              Add a Recap
            </div>
          </button>
        </div>
        <p
          class="font-sans text-gray-800 text-base "
        >
          Recap the organizations you were a part of and your impact in those groups.
        </p>
      </div>
    </header>
    <section
      class=""
    >
      <div
        class="bg-white shadow-md rounded px-8 py-6 p-4 flex mb-4"
        data-testid="organizationsRecap"
      >
        <div
          class="flex justify-center items-center p-3 rounded-lg bg-gray-200 recap-icon mr-4"
        >
          <div
            class="font-sans text-3xl text-teal-400"
            id=""
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-users fa-w-20 "
              data-icon="users"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 640 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M96 224c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm448 0c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm32 32h-64c-17.6 0-33.5 7.1-45.1 18.6 40.3 22.1 68.9 62 75.1 109.4h66c17.7 0 32-14.3 32-32v-32c0-35.3-28.7-64-64-64zm-256 0c61.9 0 112-50.1 112-112S381.9 32 320 32 208 82.1 208 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C179.6 288 128 339.6 128 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zm-223.7-13.4C161.5 263.1 145.6 256 128 256H64c-35.3 0-64 28.7-64 64v32c0 17.7 14.3 32 32 32h65.9c6.3-47.4 34.9-87.3 75.2-109.4z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
        <div
          class="flex-auto"
        >
          <h4
            class="font-mono font-bold text-gray-800 text-xl "
          >
            Zeta Mu Beta
          </h4>
          <p
            class="font-sans text-gray-800 text-base "
          >
            Long Beach, CA
          </p>
          <p
            class="font-sans text-gray-800 text-base "
          >
            Pledge Educator, Member
          </p>
          <p
            class="font-sans text-gray-800 text-base italic text-gray-600"
          >
            Dec 2014
             -
             
            Mar 2016
          </p>
          <ul
            class="mt-3"
          >
            <li
              class="flex items-center"
            >
              <div
                class="font-sans text-2xl mr-2 text-teal-300"
                id=""
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-square fa-w-14 "
                  data-icon="square"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
                    fill="currentColor"
                  />
                </svg>
              </div>
              <p
                class="font-sans text-gray-800 text-base pt-1"
              >
                Crossed into the organization in Fall 2014
              </p>
            </li>
            <li
              class="flex items-center"
            >
              <div
                class="font-sans text-2xl mr-2 text-teal-300"
                id=""
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-square fa-w-14 "
                  data-icon="square"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48z"
                    fill="currentColor"
                  />
                </svg>
              </div>
              <p
                class="font-sans text-gray-800 text-base pt-1"
              >
                Served as an active member for 2 years and as a pledge educator for the XI class
              </p>
            </li>
          </ul>
        </div>
        <div
          class="w-1/3 text-right"
        >
          <button
            class="font-bold py-2 px-4 rounded bg-transparent text-teal-700 border border-teal-500 hover:bg-teal-500 hover:text-white hover:border-transparent mr-2"
            id="organizationsId"
            type="button"
          >
            <div
              class="flex items-center"
            >
              <div
                class="font-sans text-2xl mr-2"
                id=""
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-pencil-alt fa-w-16 "
                  data-icon="pencil-alt"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
                    fill="currentColor"
                  />
                </svg>
              </div>
               Edit
            </div>
          </button>
          <button
            class="font-bold py-2 px-4 rounded font-bold py-2 px-4 rounded bg-red-500 text-white hover:bg-red-700 "
            id="organizationsId"
            type="button"
          >
            <div
              class="flex items-center"
            >
              <div
                class="font-sans text-2xl mr-2"
                id=""
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-trash-alt fa-w-14 "
                  data-icon="trash-alt"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
                    fill="currentColor"
                  />
                </svg>
              </div>
              Delete
            </div>
          </button>
        </div>
      </div>
    </section>
  </div>
</div>
`;
